src/Tribe/Views/V2/Widgets/Widget_Abstract.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Tribe\Events\Views\V2\Widgets;
+
+use Tribe\Events\Views\V2\View;
+use Tribe\Events\Views\V2\View_Interface;
+use Tribe__Context as Context;
+
+/**
+ * The abstract all widgets should implement.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Widget
+ *
+ */
+abstract class Widget_Abstract extends \Tribe\Widget\Widget_Abstract {
+
+	/**
+	 * The view interface for the widget.
+	 *
+	 * @since TBD
+	 *
+	 * @var View_Interface;
+	 */
+	protected $view;
+
+	/**
+	 * The slug of the widget view.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected $view_slug;
+
+	/**
+	 * Setup the view for the widget.
+	 *
+	 * @since TBD
+	 */
+	public function setup_view() {
+		$context = tribe_context();
+
+		// Modifies the Context for the widget params.
+		// todo update per https://github.com/moderntribe/tribe-common/pull/1451#discussion_r501498990
+		$context = $this->alter_context( $context );
+
+		// Setup the view instance.
+		$view = View::make( $this->get_view_slug(), $context );
+
+		$view->get_template()->set_values( $this->get_arguments(), false );
+
+		$this->set_view( $view );
+	}
+
+	/**
+	 * Returns the rendered View HTML code.
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function get_html() {
+		return $this->get_view()->get_html();
+	}
+
+	/**
+	 * {@inheritDoc}
```

Comment:
:no_entry_sign: Doc comment for parameter "$view" missing

<sub>hash: 143729ff81b142bef3732b0ced10f192</sub>